src/js/components/DataTable/DataTable.js

```diff
@@ -90,19 +90,29 @@ const DataTable = ({
   ...rest
 }) => {
   const theme = useContext(ThemeContext) || defaultProps.theme;
-  const { data: contextData, properties } = useContext(DataContext);
+  const { view, data: contextData, properties } = useContext(DataContext);
   const data = dataProp || contextData || emptyData;
 
   const columns = useMemo(() => {
-    if (columnsProp) return columnsProp;
-    if (properties)
-      return Object.keys(properties).map((p) => ({
+    let result = [];
+    if (columnsProp) result = columnsProp;
+    else if (properties)
+      result = Object.keys(properties).map((p) => ({
         property: p,
         ...properties[p],
       }));
-    if (data.length) return Object.keys(data[0]).map((p) => ({ property: p }));
-    return [];
-  }, [columnsProp, data, properties]);
+    else if (data.length)
+      result = Object.keys(data[0]).map((p) => ({ property: p }));
+    if (view?.columns)
+      result = result
+        .filter((c) => view.columns.includes(c.property))
+        .sort(
```

Comment:
Is there a way to make columns keep track of their last position and let them sort close to where they last were? It's confusing to hide then show a column and have it kicked to the end.